### PR TITLE
Refact: TransferWise token handler

### PR DESCRIPTION
**Pull request by kewitz**
My main objective here is to reduce the amount of `invalid_token` errors we receive on our side through automatically retrying with a renewed token.

### CODE REVIEW

File: server/lib/transferwise.ts [hunk 1]
```diff
@@ -90,55 +90,104 @@ const parseError = (
...
-  fn: Function,
+export async function getToken(connectedAccount: ConnectedAccount, refresh = false): Promise<string> {
+  // OAuth token, require us to refresh every 12 hours
+  const tokenCreation = moment.utc(connectedAccount.data.created_at);
+  const diff = moment.duration(moment.utc().diff(tokenCreation)).asSeconds();
+  const isOutdated = diff > <number>connectedAccount.data.expires_in - 60;
+  if (refresh || isOutdated) {
+    const newToken = await getOrRefreshToken({ refreshToken: connectedAccount.refreshToken });
+    if (!newToken) {
+      throw new Error('There was an error refreshing the Transferwise token');
```
Comment (hdiniz) [inline server/lib/transferwise.ts:101]: Would this require reconnecting the wise account? If so, we could send an email on this error

File: server/lib/transferwise.ts [hunk 2]
```diff
@@ -90,55 +90,104 @@ const parseError = (
...
+    )}`,
+  );
+
+  try {
+    const pRequest = data ? fn(url, data, options) : fn(url, options);
+    const response = await pRequest;
+    return getData(response);
+  } catch (e: any) {
+    const signatureFailed = e?.response?.headers?.['x-2fa-approval-result'] === 'REJECTED';
+    const hadSignature = e?.response?.headers?.['X-Signature'];
```
Comment (hdiniz) [inline server/lib/transferwise.ts:152]:
```suggestion
    const hadSignature = options?.headers?.['X-Signature'];
```

looking at what we sent in the request to decide to recurse or not.

File: server/lib/transferwise.ts (hunk 1 above)
Comment (kewitz) [reply server/lib/transferwise.ts:101]: That's a very good idea.

Comment (kewitz) [reply server/lib/transferwise.ts]: I'll start by tracking the activity to make sure that it wouldn't spam host admins.

**Pull request merged**